ares/js/components/autocomplete/autocomplete.test.js

```diff
@@ -0,0 +1,826 @@
+        );
+        expect(new_autocomplete.retreiveDestinationValueToPrePopulateInput('input#address-input')).toEqual('Austin, TX');
+    });
+});
+
+/* - - - - - - setAndClearInput - - - - - -*/
+
+describe('setAndClearInput', () => {
+    beforeAll(() => {
+        document.body.innerHTML = '';
+    });
+
+    afterAll(() => {
+        document.body.innerHTML = '';
+    });
+
+    it('Sets the value of the input with the destination', () => {
+        document.body.innerHTML = `<input type="search" id="address-input" placeholder="Destination" required="true" value="">`;
+
+        autocomplete.setAndClearInput('input#address-input', 'Austin, TX');
+
+        expect(document.querySelector('input#address-input').value).toEqual('Austin, TX');
+    });
+
+    it('On click of the input, the value is cleared', () => {
+        document.body.innerHTML = `<input type="search" id="address-input" placeholder="Destination" required="true" value="Austin, TX">`;
+
+        autocomplete.setAndClearInput('input#address-input', 'Austin, TX');
+        document.querySelector('input#address-input').click();
+        expect(document.querySelector('input#address-input').value).toEqual('');
+    });
+});
+
+/* - - - - - - appendParamsToUrl - - - - - -*/
+
+describe('appendParamsToUrl', () => {
+    beforeEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('Builds the URL with valid params', () => {
+        const url = `https://events.hotelsforhope.com/v6/?type=geo&siteid=12345&pagesize=10&useMiles=`;
+        const built_url = new URL(url);
+
+        autocomplete.appendParamsToURL(built_url, {
+            longitude: {
+                key: 'longitude',
+                value: '3',
+            },
+            rooms: {
+                key: 'rooms',
+                value: '2',
+            },
+        });
+
+        expect(built_url.toString()).toEqual(`${url}&longitude=3&rooms=2`);
+    });
+
+    it("Doesn't append any values or keys that are null, undefined or an empty string", () => {
+        const url = `https://events.hotelsforhope.com/v6/?type=geo&siteid=12345&pagesize=10&useMiles`;
+        const built_url = new URL(url);
+
+        autocomplete.appendParamsToURL(built_url, {
+            longitude: {
+                key: 'latitude',
+                value: '3',
+            },
+            properties: {
+                key: 'properties',
+                value: null,
+            },
+            adults: {
+                key: '',
+                value: '1',
+            },
+            checkin: {
+                key: 'checkin',
+                value: undefined,
+            },
+        });
+
+        expect(built_url.toString()).toEqual('https://events.hotelsforhope.com/v6/?type=geo&siteid=12345&pagesize=10&useMiles=&latitude=3');
+    });
+});
+
+/* - - - - - - setDateFormat - - - - - -*/
+
+describe('setDateFormat', () => {
```

Comment:
The static dates in this method likely need to be swapped out for something more dynamic, even if it's just incrementing the year based on the current date, unless you're 100% confident testing dates in the past still work the same as testing dates in the future in the applicable code.